codex-rs/exec-server/src/posix/escalate_protocol.rs

```diff
@@ -0,0 +1,50 @@
+use std::collections::HashMap;
+use std::os::fd::RawFd;
+use std::path::PathBuf;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+/// 'exec-server escalate' reads this to find the inherited FD for the escalate socket.
+pub(super) const ESCALATE_SOCKET_ENV_VAR: &str = "CODEX_ESCALATE_SOCKET";
+
+/// The patched bash uses this to wrap exec() calls.
+pub(super) const BASH_EXEC_WRAPPER_ENV_VAR: &str = "BASH_EXEC_WRAPPER";
+
+// C->S on the escalate socket
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub(super) enum EscalateClientMessage {
+    /// The client wants to run exec() with the given arguments.
+    EscalateRequest {
+        /// The absolute path to the executable to run, i.e. the first arg to exec.
+        file: PathBuf,
+        /// The argv, including the program name (argv[0]).
+        argv: Vec<String>,
+        workdir: PathBuf,
+        env: HashMap<String, String>,
+    },
+}
+
+// C->S on the escalate socket
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub(super) enum EscalateServerMessage {
+    EscalateResponse(EscalateAction),
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub(super) enum EscalateAction {
+    RunInSandbox,
```

Comment:
I feel like just `Run` or `Approved` is a little better because there is nothing that requires the client to be a sandbox. Similarly, escalation doesn't guarantee that it is _not_ run in a sandbox: it could very well still be a sandbox, just with different rules.